splink/profile_data.py

```diff
@@ -230,19 +233,28 @@ def profile_columns(linker, column_expressions, top_n=10, bottom_n=10):
         percentile_rows = [
             p for p in percentile_rows_all if p["group_name"] == _group_name(expression)
         ]
-        percentile_rows = _add_100_percentile_to_df_percentiles(percentile_rows)
-        top_n_rows = [
-            p for p in top_n_rows_all if p["group_name"] == _group_name(expression)
-        ]
-        bottom_n_rows = [
-            p for p in bottom_n_rows_all if p["group_name"] == _group_name(expression)
-        ]
-        # remove concat blank from expression title
-        expression = expression.replace(", ' '", "")
-        inner_chart = _get_inner_chart_spec_freq(
-            percentile_rows, top_n_rows, bottom_n_rows, expression
-        )
-        inner_charts.append(inner_chart)
+        if percentile_rows == []:
+            logger.warning(
+                "Warning: No charts produced for "
+                f"{expression}"
+                " as the column only contains null values."
+            )
+        else:
+            percentile_rows = _add_100_percentile_to_df_percentiles(percentile_rows)
+            top_n_rows = [
+                p for p in top_n_rows_all if p["group_name"] == _group_name(expression)
+            ]
+            bottom_n_rows = [
+                p
+                for p in bottom_n_rows_all
+                if p["group_name"] == _group_name(expression)
+            ]
+            # remove concat blank from expression title
+            expression = expression.replace(", ' '", "")
+            inner_chart = _get_inner_chart_spec_freq(
+                percentile_rows, top_n_rows, bottom_n_rows, expression
+            )
+            inner_charts.append(inner_chart)
     outer_spec = deepcopy(_outer_chart_spec_freq)
 
     outer_spec["vconcat"] = inner_charts
```

Comment:
At the moment, we get an odd output if the user tries to profile a single empty column...
![Screenshot 2023-06-29 at 13 41 14](https://github.com/moj-analytical-services/splink/assets/45356472/cb8c36f1-01bd-41e3-b35c-ae400eba96b4)

Perhaps it would make sense to check if there are any charts to output and then only return a result if there is?

```py
if inner_charts:
...
    return vegalite_or_json(outer_spec)
```

Reply:
I'm not sure I agree with this being an odd error message- this needs to return _something_ so the user knows it's their error, and it's that or something more general - i.e. "No charts produced due to missing data", but I think the specificity aids in de-bugging this if you've done it by accident.

Reply:
The message isn't what's odd about it. In the special case that there's only one null column (or multiple columns that are all null) then the function still attempts to produce an empty chart after printing the error message.